core/course_controller/CourseController.py

```diff
@@ -1,48 +1,226 @@
         Post-conditions: Removes a record from Course with matching course_id.
-        Side-effects: Adds new record to Course table.
+        Side-effects: removes matching record from Course table and any objects with foreign key references to it.
+        """
+        try:
+            course = Course.objects.get(course_code=course_code)
+            course.delete()
+        except Course.DoesNotExist:
+            raise ValueError("Course with the given code does not exist.")
+
+    @staticmethod
+    def save_lab_sections(course_data: CourseFormData) -> str:
+        """
+        Helper method for save course to handle adding new lab sections.
+
+        Pre-conditions: course_data is valid in the CourseFormData provided (course_code is provided).
+        Post-conditions: Adds record(s) to the LabSection table with the relation to the course_code provided in CourseFormData.
+        Side-effects: new record(s) added to the LabSection table.
+        Returns: list of id(s) of the new lab section objects added to the table or an IllegalValueException if CourseFormData
+            has no course code.
+        """
+        if not course_data.course_code:
+            raise ValueError("Course code is required for saving lab sections.")
+
+        if not course_data.lab_sections_codes:
+            # If no lab sections are provided, return early
+            return "No lab sections to add."
+
+        try:
+            course = Course.objects.get(course_code=course_data.course_code)
+        except Course.DoesNotExist:
+            raise ValueError("Course with the given code does not exist.")
+
+        existing_sections = LabSection.objects.filter(course=course).values_list('lab_section_number', flat=True)
+
+        for section_number in course_data.lab_sections_codes:
+            if section_number in existing_sections:
+                raise ValueError(f"Duplicate lab section number: {section_number}")
+            LabSection.objects.create(
+                course=course,
+                lab_section_number=section_number,
+                start_time="00:00",
+                end_time="00:00",
+            )
+
+        return f"Lab sections added to course {course_data.course_code}"
+
+    @staticmethod
+    def save_course_sections(course_data: CourseFormData) -> str:
+        """
+        Helper method for save course to handle adding new course sections.
+
+        Pre-conditions: course_data is valid in the CourseFormData provided (course_code is provided).
+        Post-conditions: Adds record(s) to the CourseSection table with the relation to the course_code provided in CourseFormData.
+        Side-effects: new record(s) added to the CourseSection table.
+        Returns: list of id(s) of the new course section objects added to the table or an IllegalValueException if CourseFormData
+            doesn't have a course_code.
         """
-        pass
\ No newline at end of file
+        if not course_data.course_code:
```

Comment:
if you pass the course_code, instructor, and course_section_codes directly then type checking will handle these error calls and unnecessary data won't be passed